asgi_redis/sentinel.py

```diff
@@ -0,0 +1,208 @@
+                capacity=capacity,
+                channel_capacity=channel_capacity,
+            )
+            # Make sure they provided some hosts, or provide a default
+            if not hosts:
+                hosts = [("localhost", 26379)]
+            self.hosts = list()
+            self.services = list()
+
+            if not services:
+                raise ValueError("Must specify at least one service name monitored by Sentinel")
+
+            if isinstance(services, six.string_types):
+                raise ValueError("Sentinel service types must be specified as an iterable list of strings")
+
+            for entry in services:
+                if not isinstance(entry, six.string_types):
+                    raise ValueError("Sentinel service types must be specified as strings.")
+                else:
+                    self.services.append(entry)
+
+            if isinstance(hosts, six.string_types):
+                # user accidentally used one host string instead of providing a list of hosts
+                raise ValueError('ASGI Redis hosts must be specified as an iterable list of hosts.')
+
+            for entry in hosts:
+                if isinstance(entry, six.string_types):
+                    raise ValueError("Sentinel Redis host entries must be specified as tuples, not strings.")
+                else:
+                    self.hosts.append(entry)
+            self.prefix = prefix
+            assert isinstance(self.prefix, six.text_type), "Prefix must be unicode"
+            # Precalculate some values for ring selection
+            self.ring_size = len(self.services)
+            # Create connections ahead of time (they won't call out just yet, but
+            # we want to connection-pool them later)
+            if socket_timeout and socket_timeout < self.blpop_timeout:
+                raise ValueError("The socket timeout must be at least %s seconds" % self.blpop_timeout)
+            self._sentinel = self._generate_sentinel(
+                redis_kwargs={
+                    "socket_connect_timeout": socket_connect_timeout,
+                    "socket_timeout": socket_timeout,
+                    "socket_keepalive": socket_keepalive,
+                    "socket_keepalive_options": socket_keepalive_options,
+                },
+            )
+            # Decide on a unique client prefix to use in ! sections
+            # TODO: ensure uniqueness better, e.g. Redis keys with SETNX
+            self.client_prefix = "".join(random.choice(string.ascii_letters) for i in range(8))
+            # Register scripts
+            connection = self.connection(None)
+            self.chansend = connection.register_script(self.lua_chansend)
+            self.lpopmany = connection.register_script(self.lua_lpopmany)
+            self.delprefix = connection.register_script(self.lua_delprefix)
+            self.incrstatcounters = connection.register_script(self.lua_incrstatcounters)
```

Comment:
Could you refactor this and the base class to bring all the common stuff like this into one place so it's not duplicated?